components/automate-ui/src/app/services/chef-session/chef-session.service.ts

```diff
@@ -61,36 +70,45 @@ export class ChefSessionService implements CanActivate {
       // and components/session-service and chef-session-service -- it would
       // make sense to try to refresh the session if we get a 401 from the API,
       // before giving up and calling logout().
-      window.setInterval(() => {
-        /* Using XMLHttpRequest instead of angular/http to avoid
-           dependency hell. */
-        const xhr = new XMLHttpRequest();
-        xhr.onreadystatechange = this.refreshSessionCallback.bind(this);
-        xhr.open('GET', '/session/refresh', true);
-        xhr.responseType = 'json';
-        xhr.setRequestHeader('Authorization', `Bearer ${this.id_token}`);
-        xhr.send(null);
-      }, minute);
+      timer(0, minute).pipe(
+        filter(_ => !this.isRefreshing),
+        mergeMap(_ => {
```

Comment:
Convention for "no args" in each case:
```suggestion
        filter(() => !this.isRefreshing),
        mergeMap(() => {
```